mongodb/intro/querying-arrays/querying-an-array-with-the-all-operator-iv.md

```diff
@@ -0,0 +1,166 @@
+---
+author: Stefan-Stojanovic
+
+aspects:
+  - introduction
+
+type: normal
+
+category: must-know
+
+links:
+
+---
+# Querying An Array With The $all Operator
+---
+## Content
+
+Now that we know how to query an array for a single value, let's increase the value count and add additional query operators for a more precise query.
+
+Querying the same database as before [1].
+
+```js
+db.pokedex.find({
+  spells: ["Seed Bomb", "Bite", "Hydro-Pump"]
+});
+```
+The above method searches for any documents that contain the `spells` field and looks only for the documents that have "Seed Bomb" as the first value, "Bite" as the second and "Hydro-Pump" as the third value. Giving us the output:
+
+```javascript
+{
+  _id: ObjectId("5d9d8a6a0b24990f19398209"),
+  name: "Bulbasaur",
+  type: "Grass",
+  power: 311,
+  spells: ["Seed Bomb", "Bite", "Hydro-Pump"]
+},
+```
+
+If we wanted to search which array contains the above 3 values, regardless of the order they are in, we would do so by adding the `$all` operator. 
+
+**Note:** The `$all` operator behaves the same as the `$and` operator(discussed in the `Querying Operators` workout).
```

Comment:
```suggestion
**Note:** The `$all` operator behaves the same as the `$and` operator (discussed in the `Querying Operators` workout).
```

Reply:
![Capture](https://user-images.githubusercontent.com/22899166/70907021-0ab9d700-2008-11ea-93f3-03d111bfa217.PNG)

Reply:
Sure but the operators don't behave the same, they just have overlapping behavior.

Reply:
Saying operators behave the same means that we can use `$all` or `$and` interchangeably, which isn't the case.

Reply:
True, I'll commit the suggested change

Reply:
I'd go even further and maybe say:

```suggestion
**Note:** A query written with the `$all` operator can also be written with the `$and` operator (as discussed in the `Querying Operators` workout).
```

What do you think?